test/include/concurrency/transaction_tests_util.h

```diff
@@ -332,13 +341,18 @@ class TransactionThread {
 class TransactionScheduler {
  public:
   TransactionScheduler(size_t num_txn, storage::DataTable *datatable_,
-                       concurrency::TransactionManager *txn_manager_)
+                       concurrency::TransactionManager *txn_manager_,
+                       bool first_as_ro = false)
       : txn_manager(txn_manager_),
         table(datatable_),
         time(0),
         concurrent(false) {
     for (size_t i = 0; i < num_txn; i++) {
-      schedules.emplace_back(i);
+      if (first_as_ro && i == 0) {
```

Comment:
Why only the first transaction can be read_only? Is that a requirement for batched transactions in our system?